### ISSUE TITLE
Tenant does not have a SPO license.

### ISSUE THREAD

**Issue by RobertStaflin** [laurent22/joplin#5995]
I have Joplin installed on around 7 different machines, Mac and Windows, all synced to OneDrive. On my main machine (a MacBook Pro M1 Max), I get this error since a few weeks: 

`Error: Could not retrieve account details (drive ID, Account type. Error code: BadRequest, Error message: Tenant does not have a SPO license.`

I have deleted and reinstalled Joplin, deleted the application settings, and tried deleting and resyncing the Joplin library (where everything is saved in OneDrive, i.e. ~/OneDrive/Apps/Joplin

## Environment

Joplin version: 2.6.10
Platform: macOS
OS specifics: macOS 12.1

## Steps to reproduce

1. On this machine: install Joplin 
2. Start Joplin
3. Observe the error message
4. ...on my other machines everything just works.

## Describe what you expected to happen

Syncing that works as it does on the other machines. Since it's my main machine that has the problem, I'm thinking of stopping usage of Joplin entirely

## Logfile

17:17:12: Could not run background sync:
Logger.ts:217 17:17:12: Error: Could not retrieve account details (drive ID, Account type. Error code: BadRequest, Error message: Tenant does not have a SPO license.
    at OneDriveApi.<anonymous> (onedrive-api.ts:381)
    at Generator.throw (<anonymous>)
    at rejected (/Applications/Joplin.app/Contents/Resources/app.asar/node_modules/@joplin/lib/onedrive-api.js:6)
    at processTicksAndRejections (internal/process/task_queues.js:95)

**Comment by leematos** [comment]
hmmm. What are the versions you are using across all of your devices? Are they all 2.6.10?

**Comment by RobertStaflin** [comment]
Yes, now they are, but they might not have been when the problem started. If I could only find a way of starting everything over for all the clients… 🤔 Perhaps I could do a backup, and then remove the OneDrive folder entirely…

Skickat från min iPad

**Comment by RobertStaflin** [comment]
I’m switching over to iCloud/Filesystem syncing, so don’t worry about me. :-)

**Comment by RobertStaflin** [comment]
> I saw this stackoverflow: https://stackoverflow.com/a/55036124 are you using a business/personal account?

I am using both, in an "unholy mix". For my personal/business use I have one account (the same account signed in to both business och personal), and for work I'm also signed in to MS365 with another account.

**Comment by RobertStaflin** [comment]
I think we can close this. https://www.computerworld.com/article/3648044/dropbox-microsoft-onedrive-send-out-warnings-to-macos-users.html

...so I think this might be it.
